File: demo-application/src/main/java/de/jds/DemoSynConfigurer.java
package de.jds;

import org.springframework.context.annotation.Configuration;
import org.springframework.sync.diffsync.PersistenceCallbackRegistry;
import org.springframwork.sync.config.DiffSyncConfigurer;

@Configuration
public class DemoSynConfigurer implements DiffSyncConfigurer {
  @Override
  public void addPersistenceCallbacks(final PersistenceCallbackRegistry persistenceCallbackRegistry) {
    persistenceCallbackRegistry.addPersistenceCallback(new DemoPersistenceCallback());
  }
}

File: demo-application/src/main/kotlin/de/jds/example/DemoSynConfigurer.kt
package de.jds.example

import org.springframework.context.annotation.Configuration
import org.springframework.sync.persistence.PersistenceCallbackRegistry

@Configuration
class DemoSynConfigurer : DiffSyncConfigurer {
    override fun addPersistenceCallbacks(persistenceCallbackRegistry: PersistenceCallbackRegistry) {
        persistenceCallbackRegistry.addPersistenceCallback(DemoPersistenceCallback())
    }
}